astropy/io/misc/parquet.py

```diff
@@ -0,0 +1,313 @@
+    else:
+        num_rows = 0
+
+    dtype = []
+    for name in columns_to_read:
+        if schema.field(name).type == pa.string() or schema.field(name).type == pa.binary():
+            md_name = f'table::len::{name}'
+            if md_name in md:
+                # String/bytes length from header.
+                strlen = int(md[f'table::len::{name}'])
+            else:
+                # Find the maximum string length.
+                if schema_only:
+                    # Choose an arbitrary string length since
+                    # are not reading in the table.
+                    strlen = 10
+                    warnings.warn(f"No table::len::{name} found in metadata. "
+                                  f"Guessing {{strlen}} for schema.",
+                                  AstropyUserWarning)
+                else:
+                    strlen = max([len(row.as_py()) for row in pa_table[name]])
+                    warnings.warn(f"No table::len::{name} found in metadata. "
+                                  f"Using longest string ({{strlen}} characters).",
+                                  AstropyUserWarning)
+            if schema.field(name).type == pa.string():
+                dtype.append(f'U{strlen}')
+            else:
+                dtype.append(f'|S{strlen}')
+        else:
+            dtype.append(schema.field(name).type.to_pandas_dtype())
+
+    data = np.zeros(num_rows, dtype=list(zip(columns_to_read, dtype)))
+
+    if not schema_only:
+        for name in columns_to_read:
+            data[name][:] = pa_table[name].to_numpy()
+
+    table = Table(data=data)
+    if meta_dict != {}:
+        table.meta = meta_dict
+
+    if meta_hdr is not None:
+        header_cols = dict((x['name'], x) for x in meta_hdr['datatype'])
+        for col in table.columns.values():
+            for attr in ('description', 'format', 'unit', 'meta'):
+                if attr in header_cols[col.name]:
+                    setattr(col, attr, header_cols[col.name][attr])
+
+    table = serialize._construct_mixins_from_columns(table)
+
+    return table
+
+
+def write_table_parquet(table, output, overwrite=False):
+    """
+    Write a Table object to a Parquet file
+
+    This requires `pyarrow <https://arrow.apache.org/docs/python/>`_
+    to be installed.
+
+    Parameters
+    ----------
+    table : `~astropy.table.Table`
+        Data table that is to be written to file.
+    output : str
+        The filename to write the table to.
+    overwrite : bool
+        Whether to overwrite any existing file without warning.
+    """
+
+    from astropy.table import meta, serialize
+    from astropy.utils.data_info import serialize_context_as
+    try:
+        import pyarrow as pa
+        from pyarrow import parquet
+    except ImportError:
+        raise Exception("pyarrow is required to read and write Parquet files")
+
+    if not isinstance(output, str):
+        raise TypeError('output should be a string')
```

Comment:
Should probably indicate `output` is an input argument. Also dump some more info.

```suggestion
        raise TypeError(f'`output` should be a string, not {output}')
```